packages/pluggableWidgets/video-player-web/src/components/SizeContainer.tsx

```diff
@@ -21,59 +23,67 @@ export interface SizeProps extends Dimensions {
 
 export class SizeContainer extends Component<SizeProps> {
     render(): JSX.Element {
-        const styleWidth = this.props.widthUnit === "percentage" ? `${this.props.width}%` : `${this.props.width}px`;
-        const relativeStyle: CSSProperties = {
-            position: "relative",
-            width: styleWidth,
-            ...this.getHeight(this.props.heightUnit, this.props.height, this.props.widthUnit, this.props.width),
-            display: "flex",
-            justifyContent: "center",
-            ...this.props.style
-        };
-        const absoluteStyle: CSSProperties = {
-            position: "absolute",
-            top: "0",
-            right: "0",
-            bottom: "0",
-            left: "0",
-            display: "flex",
-            justifyContent: "center"
-        };
-
         return (
             <div
                 className={classNames(this.props.className, "size-box")}
-                style={relativeStyle}
                 tabIndex={this.props.tabIndex}
+                style={{ ...this.setDimensions(), ...this.props.style }}
             >
-                <div className={classNames("size-box-inner", this.props.classNameInner)} style={absoluteStyle}>
-                    {this.props.children}
-                </div>
+                <div className={classNames("size-box-inner", this.props.classNameInner)}>{this.props.children}</div>
             </div>
         );
     }
 
-    private getHeight(
-        heightUnit: HeightUnitType,
-        height: number,
-        widthUnit: WidthUnitType,
-        width: number
-    ): CSSProperties {
+    private setDimensions(): CSSProperties {
+        const { widthUnit, width, heightUnit, height, heightAspectRatio } = this.props;
+
         const style: CSSProperties = {};
-        if (heightUnit === "percentageOfWidth") {
-            const ratio = (height / 100) * width;
-            if (widthUnit === "percentage") {
-                style.height = "auto";
-                style.paddingBottom = `${ratio}%`;
-            } else {
-                style.height = `${ratio}px`;
-            }
-        } else if (heightUnit === "pixels") {
-            style.height = `${height}px`;
-        } else if (heightUnit === "percentageOfParent") {
-            style.height = `${height}%`;
-        } else if (heightUnit === "aspectRatio") {
-            style.height = width * 0.5625; // Default is 16:9
+        style.width = widthUnit === "percentage" ? `${width}%` : `${width}px`;
+
+        switch (heightUnit) {
+            case "pixels":
+                style.paddingTop = height;
+                break;
+            case "percentageOfWidth":
+                const actualHeight = (height! / 100) * width;
+
+                if (widthUnit === "percentage") {
+                    style.paddingTop = `${actualHeight}%`;
+                } else {
+                    style.paddingTop = actualHeight;
+                }
```

Comment:
same thing here, think this should just be 
```
style.paddingTop = `{height}%`
```

Reply:
See [this](https://github.com/mendix/widgets-resources/pull/686#discussion_r661541319) answer.